### PR TITLE
[ENH] Tidy up and restructure load to nested univariate

### PR DESCRIPTION
**Pull request by TonyBagnall**
Part of #974 

This simplifies and tidies up the file datasets._dataframe_loaders.py. Context is this is very old code to load data into the now defunct nested univ format. It is not used anywhere and has under 50% coverage. Changes:

1. Remove load from long format to nested univariate. This function makes a lot of assumptions about the input format and has no facility to allow class values. I see no point in keeping it
2. Load the tsv, ts and arff load to dataframe functions simply call their counterparts in _data_loaders, then convert to nested_univ. Removes a lot of code and makes it all more readable and robust. It does mean reducing some of the options in the arguments, but these were not tested anywhere and seemed pointless to me. 
3. Tidies up a weird potential circular import I left in when writing convert_collection
4. Removes a function load_tsf_to_dataframe from _data_loaders that was a direct copy of the function in _dataframe_loaders
5. Remove the duplication of having load_from_tsf_file and load_tsf_to_dataframe, because they were essentially doing the same thing. Just have load_from_tsf_file
6. Adds some tests for _data_loaders
7. Make separate test files for _data_generators and forecasting loading, to make things clearer 
8. Adjusted tsf single problem loader to return either default format or the hierarchical long format.

Two issues for me need resolving, possibly after this PR
1. _data_generators contains a function _convert_tsf_to_hierarchical which should be elsewhere really
2. Im still unclear as to the best format for returning time series for forecasting, need to explore a little more

### CODE REVIEW

File: aeon/datasets/_dataframe_loaders.py [hunk 1]
```diff
@@ -615,110 +61,27 @@ def load_from_arff_to_dataframe(
     return_separate_X_and_y=True,
```
Comment (MatthewMiddlehurst) [inline aeon/datasets/_dataframe_loaders.py:61]: Unused parameters again

File: aeon/datasets/tests/test_dataframe_loaders.py [hunk 2]
```diff
@@ -16,10 +23,357 @@
...
+        "contain_equal_length": False,
+    }
+
+    df, metadata = load_tsf_to_dataframe(data_path, return_type=return_type)
+    assert isinstance(df, pd.DataFrame)
+    assert isinstance(metadata, dict)
+    assert_frame_equal(df, output_df, check_dtype=False)
+    assert metadata == expected_metadata
+    # default_tsf"
+    #    assert check_is_mtype(obj=df, mtype=return_type)
```
Comment (MatthewMiddlehurst) [inline aeon/datasets/tests/test_dataframe_loaders.py:373]: strange comment can probably go.

Im amazed this dosent fail given the conversion bits are removed, does the function before already set it to these types?

Comment (TonyBagnall) [reply aeon/datasets/tests/test_dataframe_loaders.py:373]: the only conversion removed is from pd_multiindex_hier to pd_multiindex (_convert_tsf_to_hierarchical returns  pd_multiindex_hier) and the distinction between the two is not actually tested. If anyone ever wanted such a distinction we could put it in, but I dont see the need, I think let users sort their datatypes out really!

File: aeon/datasets/_data_loaders.py [hunk 3]
```diff
@@ -11,14 +11,17 @@
...
+import aeon
 from aeon.datasets.dataset_collections import (
     list_downloaded_tsc_tsr_datasets,
     list_downloaded_tsf_datasets,
 )
-from aeon.datatypes import MTYPE_LIST_HIERARCHICAL, convert
 from aeon.utils.validation.collection import convert_collection
 
+DIRNAME = "data"
+MODULE = os.path.join(os.path.dirname(aeon.__file__), "datasets")
```
[the remark this answers is not in the document]
Comment (TonyBagnall) [reply aeon/datasets/_data_loaders.py:22]: too tedious to tidy up this legacy code imo

**Review by MatthewMiddlehurst — APPROVED** [review]

**Pull request merged**